engines/order_management/app/services/order_management/order/stripe_sca_payment_authorize.rb

```diff
@@ -1,20 +1,27 @@
 # frozen_string_literal: true
 
 module OrderManagement
-  module Subscriptions
+  module Order
     class StripeScaPaymentAuthorize
+      include FullUrlHelper
+
       def initialize(order)
         @order = order
         @payment = OrderPaymentFinder.new(@order).last_pending_payment
       end
 
-      def call!
+      def call!(redirect_url = full_order_path(@order), send_emails = true)
```

Comment:
IMO this extra field that controls the notification behavior adds an unnecessary extra responsibility to this class coupling it with the details of the mail notifications (implemented on line 20 below), and it doesn't belong here. It would increase the changeability of this implementation to move this responsibility somewhere else, more context-specific. Ideally, calling a new separate abstraction from where we initially call `StripeRedirect`, which I believe is `CheckoutController`.

I'm happy to move this concern to an issue and tackle it together. IMO it's important to have a very changeable Stripe integration if we want to successfully deal with potential bugs in the future without increasing too much their dev cost.

Reply:
I moved this to a separate module in ef6d1a3afb74854d73b74d94cb0ab8d1ab923ce1. It's not perfect, since we're still using instance variables without much context, but I think it's preferable to the flag param. Note that the Checkout flow is not unchanged, and it's the SubscriptionConfirmJob that now uses the module.